src/games/strategy/engine/framework/startup/mc/ClientModel.java

```diff
@@ -335,35 +335,38 @@ private void startGameInNewThread(final byte[] gameData, final Map<String, INode
     final Set<IGamePlayer> playerSet = data.getGameLoader().createPlayers(playerMapping);
     final Messengers messengers = new Messengers(m_messenger, m_remoteMessenger, m_channelMessenger);
     m_game = new ClientGame(data, playerSet, players, messengers);
-    final Thread t = new Thread("Client Game Launcher") {
-      @Override
-      public void run() {
-        SwingUtilities.invokeLater(() -> JOptionPane.getFrameForComponent(m_ui).setVisible(false));
-        try {
-          // game will be null if we loose the connection
-          if (m_game != null) {
-            try {
-              data.getGameLoader().startGame(m_game, playerSet, false);
-              data.testLocksOnRead();
-            } catch (final Exception e) {
-              ClientLogger.logQuietly(e);
+    new Thread(() -> {
+      SwingUtilities.invokeLater(() -> JOptionPane.getFrameForComponent(m_ui).setVisible(false));
+      try {
+        // game will be null if we loose the connection
+        if (m_game != null) {
+          try {
+            data.getGameLoader().startGame(m_game, playerSet, false, () -> {
```

Comment:
Brownie points if you replace `false` with an enum. Having looked at the method API just a moment ago, already I've forgotten if it was 'isHeadless', or 'isClient'. APIs that require you to look at their implementation/signature/javadocs to simply use them, are not usable, and scale poorly when it comes to maintenance cost. Namely whenever anyone reads this line, you have to jump around to somewhere else to understand it. Then repeat for each of the following lines.. who knows what exactly happens in game shutdown vs messenger shutdown and loading window done wait..